bytecomp/emitcode.ml

```diff
@@ -168,6 +168,37 @@ and slot_for_c_prim name =
 let events = ref ([] : debug_event list)
 let debug_dirs = ref String.Set.empty
 
+(* Map from absolute paths of .ml files to their sanitized paths. *)
+let sanitize_map = Hashtbl.create 17
+
+let sanitize_event ev =
+  let loc = ev.ev_loc in
+  let sloc = loc.loc_start in
+  let path = sloc.pos_fname in
+  let open Filename in
+  let unmapped_abspath =
+    if (is_relative path) then (concat (Sys.getcwd ()) path)
+    else path
+  in
+  let mapped_abspath = Location.rewrite_absolute_path unmapped_abspath in
```

Comment:
Isn't this equivalent to `let mapped_abspath = Location.absolute_path path`?

Reply:
Yes, but I wanted the unmapped path available so it could be compared to the mapped path. If the mapping did nothing, I was just going to keep the existing event.